### PR TITLE
Added form to add a new city, added banner to the form, tested it

### PR DESCRIPTION
**Pull request by sfrunza**
I added a add new city form and also added a column "banner" in the city migration
wrote tests for the form

### CODE REVIEW

**Comment by sfrunza** [comment]
Hey Dan we get this migration failing message from Codeship, im not really sure how to fix it

File: app/controllers/cities_controller.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+class CitiesController < ApplicationController
+  def index
+    @cities = City.all
+  end
+
+  def new
+    if user_signed_in?
+      @city = City.new
+    else
+      flash[:notice] = "You need to be signed in to add a City."
```
Comment (dpickett) [inline app/controllers/cities_controller.rb:10]: good!

File: app/controllers/cities_controller.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      redirect_to new_user_session_path
+    end
+  end
+
+  def create
+    @city = City.new(city_params.merge(user_id: current_user.id))
+    unless user_signed_in?
```
Comment (dpickett) [inline app/controllers/cities_controller.rb]: we should only run the conditional below *if* the user is signed in. Execution will not stop on the `redirect`

File: spec/features/user_can_add_a_city.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  end
+
+  scenario "user does not provide proper information for a city" do
+    visit new_city_path
+
+    click_button "Add City"
+    expect(page).to have_content "Name can't be blank"
+    expect(page).to have_content "State can't be blank"
+
+  end
```
Comment (dpickett) [inline spec/features/user_can_add_a_city.rb:29]: we should also have a scenario where a signed out user cannot add a city.

File: spec/features/user_can_add_a_city.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+end
+describe City do
+  include CarrierWave::Test::Matchers
+    let(:user) { double('user') }
+    let(:uploader) { City.new(user, :banner) }
+    before do
+      User.enable_processing = true
+      File.open(path_to_file) { |f| uploader.store!(f) }
+    end
+    after do
```
Comment (dpickett) [inline spec/features/user_can_add_a_city.rb]: this test does not assert anything.

**Review by dpickett — CHANGES_REQUESTED** [review]

**Comment by sfrunza** [comment]
made some changes in CityTile test file and added package-lock and yarn

**Comment by sfrunza** [comment]
deleted the picture

**Pull request merged**